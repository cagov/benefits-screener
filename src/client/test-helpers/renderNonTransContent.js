import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
