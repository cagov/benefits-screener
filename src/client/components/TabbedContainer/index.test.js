import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<TabbedContainer />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component);

    expect(wrapper).toMatchSnapshot();
  });
});
