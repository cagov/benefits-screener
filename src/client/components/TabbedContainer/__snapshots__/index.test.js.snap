// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabbedContainer /> renders vertical navigation bar 1`] = `
<Container
  fluid={false}
>
  <TabContainer
    activeKey={0}
    defaultActiveKey="0"
    id="left-tabs"
  >
    <Row
      className="tabbed-container"
      noGutters={false}
    >
      <Col
        className="TabbedContainer"
        sm={4}
      >
        <Nav
          className="flex-column sidebar-sticky"
          fill={false}
          justify={false}
          variant="pills"
        >
          <NavItem
            key="0"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={0}
              href="#benefits"
              onClick={[Function]}
            >
              tabTitles.0
            </NavLink>
          </NavItem>
          <NavItem
            key="1"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={1}
              href="#before-you-apply"
              onClick={[Function]}
            >
              tabTitles.1
            </NavLink>
          </NavItem>
          <NavItem
            key="2"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={2}
              href="#how-to-apply"
              onClick={[Function]}
            >
              tabTitles.2
            </NavLink>
          </NavItem>
          <NavItem
            key="3"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={3}
              href="#after-you-submit"
              onClick={[Function]}
            >
              tabTitles.3
            </NavLink>
          </NavItem>
          <NavItem
            key="4"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={4}
              href="#receive-benefits"
              onClick={[Function]}
            >
              tabTitles.4
            </NavLink>
          </NavItem>
          <NavItem
            key="5"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              eventKey={5}
              href="#more-resources"
              onClick={[Function]}
            >
              tabTitles.5
            </NavLink>
          </NavItem>
        </Nav>
      </Col>
      <Col
        sm={8}
      >
        <ForwardRef>
          <TabPane
            eventKey={0}
            key="0"
          >
            <h2>
              tabTitles.0
            </h2>
            <TabPaneContent0
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
            <Button
              active={false}
              disabled={false}
              href="#before-you-apply"
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              buttonNextPrefixtabTitles.1
            </Button>
          </TabPane>
          <TabPane
            eventKey={1}
            key="1"
          >
            <h2>
              tabTitles.1
            </h2>
            <TabPaneContent1
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
            <Button
              active={false}
              disabled={false}
              href="#how-to-apply"
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              buttonNextPrefixtabTitles.2
            </Button>
          </TabPane>
          <TabPane
            eventKey={2}
            key="2"
          >
            <h2>
              tabTitles.2
            </h2>
            <TabPaneContent2
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
            <Button
              active={false}
              disabled={false}
              href="#after-you-submit"
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              buttonNextPrefixtabTitles.3
            </Button>
          </TabPane>
          <TabPane
            eventKey={3}
            key="3"
          >
            <h2>
              tabTitles.3
            </h2>
            <TabPaneContent3
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
            <Button
              active={false}
              disabled={false}
              href="#receive-benefits"
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              buttonNextPrefixtabTitles.4
            </Button>
          </TabPane>
          <TabPane
            eventKey={4}
            key="4"
          >
            <h2>
              tabTitles.4
            </h2>
            <TabPaneContent4
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
            <Button
              active={false}
              disabled={false}
              href="#more-resources"
              onClick={[Function]}
              type="button"
              variant="secondary"
            >
              buttonNextPrefixtabTitles.5
            </Button>
          </TabPane>
          <TabPane
            eventKey={5}
            key="5"
          >
            <h2>
              tabTitles.5
            </h2>
            <TabPaneContent5
              getTabTitle={[Function]}
              loadTab={[Function]}
              tabSlugs={
                Array [
                  "benefits",
                  "before-you-apply",
                  "how-to-apply",
                  "after-you-submit",
                  "receive-benefits",
                  "more-resources",
                ]
              }
            />
          </TabPane>
        </ForwardRef>
      </Col>
    </Row>
  </TabContainer>
</Container>
`;
